### DISCUSSION TITLE
Errror when Andoid APP starts

### DISCUSSION THREAD

**Discussion by afyfe** [indrawp/Setup-Guide#56] [Q&A]
Hi, I believe I have now a working Signal Server

But when trying to get the Andorid app working, 

Now facing this error:

```
Error during account registration 
org.whispersystems.signalservice.api.push.exceptions.DeprecatedVersionException
```

I cannot see anything else wrong.. has anyone had this issue ?

**Comment by madeindra** [comment]
@afyfe The API version is too old, which Android version did you use? Could try older version android

**Comment by afyfe** [comment]
Which api would that be ?

**Reply by madeindra** [reply]
The API that the android version called could be different with the API exist on the server side, this is normal, because the android version is being developed but the server is never updated by the signal team

**Comment by afyfe** [comment]
Is there any way to know which android version will work with the version
of signal ?

**Reply by madeindra** [reply]
You can see from the commit date, the server's latest update is on  April 23 2020, on that date the Android version was 4.59.1.

By following the rule of semantic versioning, the most compatible android version could be the v4.xx

**Comment by afyfe** [comment]
Thanks for the help so far!

We’ve done that now facing  this error :

invalidkeyexception bad key type 176


Again thanks for the help so far !

A

**Reply by madeindra** [reply]
Is this the server's error message?

**Comment by afyfe** [comment]
That’s in the Android app

**Comment by afyfe** [comment]
Full error text is ::


E/AndroidRuntime: FATAL EXCEPTION: signal-JobRunner-1
    Process: org.android.vpme, PID: 20263
    java.lang.AssertionError:
org.whispersystems.libsignal.InvalidKeyException: Bad key type: 176
        at
org.android.vpme.crypto.UnidentifiedAccessUtil.getCertificateValidator(UnidentifiedAccessUtil.java:36)
        at
org.android.vpme.jobs.PushDecryptMessageJob.handleMessage(PushDecryptMessageJob.java:167)
        at
org.android.vpme.jobs.PushDecryptMessageJob.onRun(PushDecryptMessageJob.java:113)
        at org.android.vpme.jobs.BaseJob.run(BaseJob.java:25)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:86)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:49)
     Caused by: org.whispersystems.libsignal.InvalidKeyException: Bad key
type: 176
        at org.whispersystems.libsignal.ecc.Curve.decodePoint(Curve.java:49)
        at
org.android.vpme.crypto.UnidentifiedAccessUtil.getCertificateValidator(UnidentifiedAccessUtil.java:33)
        at
org.android.vpme.jobs.PushDecryptMessageJob.handleMessage(PushDecryptMessageJob.java:167)
        at
org.android.vpme.jobs.PushDecryptMessageJob.onRun(PushDecryptMessageJob.java:113)
        at org.android.vpme.jobs.BaseJob.run(BaseJob.java:25)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:86)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:49)
E/SignalUncaughtExceptionHandler: java.lang.AssertionError:
org.whispersystems.libsignal.InvalidKeyException: Bad key type: 176
        at
org.android.vpme.crypto.UnidentifiedAccessUtil.getCertificateValidator(UnidentifiedAccessUtil.java:36)
        at
org.android.vpme.jobs.PushDecryptMessageJob.handleMessage(PushDecryptMessageJob.java:167)
        at
org.android.vpme.jobs.PushDecryptMessageJob.onRun(PushDecryptMessageJob.java:113)
        at org.android.vpme.jobs.BaseJob.run(BaseJob.java:25)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:86)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:49)
     Caused by: org.whispersystems.libsignal.InvalidKeyException: Bad key
type: 176
        at org.whispersystems.libsignal.ecc.Curve.decodePoint(Curve.java:49)
        at
org.android.vpme.crypto.UnidentifiedAccessUtil.getCertificateValidator(UnidentifiedAccessUtil.java:33)
        at
org.android.vpme.jobs.PushDecryptMessageJob.handleMessage(PushDecryptMessageJob.java:167)
        at
org.android.vpme.jobs.PushDecryptMessageJob.onRun(PushDecryptMessageJob.java:113)
        at org.android.vpme.jobs.BaseJob.run(BaseJob.java:25)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:86)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:49)

**Reply by madeindra** [reply]
I think there is an error in your build.gradle related to key, have you generated unidentified sender key properly?

**Comment by afyfe** [comment]
we seem to have past that by downgrading like you said  - good spot!

now we face a 500 error

in nginx logs we see:

connect() failed (111: Connection refused) while connecting to upstream,
client: x.x.x.x, server: domain.com, request: "GET
/v1/websocket/?login=null&password=null HTTP/1.1", upstream:
"http://[::1]:8080/v1/websocket/?login=null&password=null",
host: hostname

any clues ?

**Reply by madeindra** [reply]
websocket connection with login & password `null` can be ignored in my experienced, it is normal since the account has not been registered.

Can you register in the app?

**Comment by afyfe** [comment]
anyone ? 

really need help with this !

many thanks in advance

**Comment by afyfe** [comment]
No, the app gives a http 500 error

**Reply by madeindra** [reply]
In the server for the registration API? or an error shown in the app?

**Comment by afyfe** [comment]
That’s shown in the app and u can also see it in the signal server log file

**Reply by madeindra** [reply]
Can you share the screenshot for both? it would be helpful for the debugging process

**Comment by afyfe** [comment]
Here is the android error attached

**Reply by madeindra** [reply]
Sorry, can't see any attachment

**Comment by afyfe** [comment]
Try this

**Reply by madeindra** [reply]
Still no

**Reply by afyfe** [reply]
See this?
![AFC63B31-E54D-45F3-8965-C76D0F6191AE](https://user-images.githubusercontent.com/1941804/114143310-e6c08a00-990b-11eb-83a4-fb3ddb4e3a5c.jpeg)

**Reply by madeindra** [reply]
Yeah saw it

**Comment by afyfe** [comment]
Did you see it now?

**Comment by afyfe** [comment]
the resultant error in the app is sending the OTP results in unable to
connect to service

**Reply by madeindra** [reply]
That would be your server http in build.gradle that have some error.

Have you tried opening this url in browser? Change the +xxxxx with your phone number in international format

```
https://yourdomain.com/v1/accounts/sms/code/+xxxxx?client=android-ng
```

**Comment by afyfe** [comment]
Hi, ok, weve got so much further now...

all errors fixed and the app is now working !!!

BUT, the chat window is blank.. so i can send and receive a message,but the
chat window doesnt show any messages at all.

thats a new one !

any clues ?!

adam

**Reply by madeindra** [reply]
I've never know about such bug, can you share the screenshot or video?

**Comment by afyfe** [comment]
Please see screenshot attached...

messages are sent and received.

ive checked the disappearing messages function, thats off.

As an aside, i looked in the Postgres database messagedb - should that be
populated with messages  ??? as its currently blank.


almost there i think!!

thanks

adam

**Reply by madeindra** [reply]
Can't see image, maybe you should post image elsewhere then share the link here

**Comment by afyfe** [comment]
![Uploading 6D7ED927-818D-44AB-BAB2-2347FE12E96D.jpeg…]()

**Reply by madeindra** [reply]
Sorry, still can't see this. Not sure it's just me or github.

**Reply by afyfe** [reply]
![Signal Adam](https://user-images.githubusercontent.com/1941804/114304320-66945300-9aca-11eb-8aaf-95978cacf9ad.JPG)

**Comment by afyfe** [comment]
Check now I’ve uploaded direct to hit hub

**Reply by madeindra** [reply]
Saw it. So each message sent does not shown there right?
When it happened, did it show an error on the logcat / server api call?

**Comment by afyfe** [comment]
The messages do get sent and received just don’t show once you’ve read them
they appear in notifications but not the chat window

**Reply by madeindra** [reply]
In that case, it would be better to ask directly to Signal Community, you might found a bug and that would help them

**Comment by afyfe** [comment]
Seems it was a version issue - messages now working
But when attached anythng the attachment never gets received. In. The config I see the gcpattachmets anf the aws one. 

Which one is actually used ? 

How can we debug to see what’s happening ? 

Thanks as ever

**Reply by madeindra** [reply]
Both are used, you can see the logcat in android & see what process fail

**Comment by afyfe** [comment]
logcat shows this:

I/BaseJob: [JOB::b9fdbf7f-b094-4833-a676-33f1976b4bf7][AttachmentUploadJob]
Encountered a retryable exception. (Time Since Submission: 9127 ms,
Lifespan: 86400000 ms, Run Attempt: 3/Unlimited)

org.whispersystems.signalservice.api.push.exceptions.NonSuccessfulResponseCodeException:
Response: Response{protocol=http/1.1, code=405, message=Method Not Allowed,
url=https://d3j5a4yh18hjqa.cloudfront.net/attachments/}
        at
org.whispersystems.signalservice.internal.push.PushServiceSocket.uploadToCdn0(PushServiceSocket.java:1152)
        at
org.whispersystems.signalservice.internal.push.PushServiceSocket.uploadAttachment(PushServiceSocket.java:982)
        at
org.whispersystems.signalservice.api.SignalServiceMessageSender.uploadAttachmentV2(SignalServiceMessageSender.java:409)
        at
org.whispersystems.signalservice.api.SignalServiceMessageSender.uploadAttachment(SignalServiceMessageSender.java:387)
        at
org.android.vpme.jobs.AttachmentUploadJob.onRun(AttachmentUploadJob.java:128)
        at org.android.vpme.jobs.BaseJob.run(BaseJob.java:25)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:86)
        at org.android.vpme.jobmanager.JobRunner.run(JobRunner.java:49)

**Reply by madeindra** [reply]
Yes it's 405, could be the server does not allow POST/GET method, try checking bucket permission

**Comment by afyfe** [comment]
hi still at this!

enabling logging on the AWS side, and this is the error when trying to send an attachment

REST.POST.OBJECT attachments/ "POST /attachments/ HTTP/1.1" 405 MethodNotAllowed 348 - 6 - "-" "Amazon CloudFront" -

and the permissions i think are correct:

![image](https://user-images.githubusercontent.com/1941804/115276694-40f6f180-a13b-11eb-80b3-ac3534a8cee7.png)


thanks for your help

a

**Reply by madeindra** [reply]
There's could be an additional policy needed by AWS, as this section of [Signal Android](https://github.com/madeindra/signal-setup-guide/blob/master/signal-server-3.21/MINIO.md#signal-android) but I'm not sure what was missing

**Comment by afyfe** [comment]
Deleting the bucket an recreating sorted it !

Last question :) how does one change the text that’s sent with the OTP
that’s sent to verify the phone number ?

**Reply by madeindra** [reply]
This is on server v3.21, isn't it? If so, you can find the string/text inside a java class, but after editing it, you need to recompile it.

In v5.xx this have been refactored and can be changed via config.yml

**Comment by afyfe** [comment]
Thank you!

Is it worth building a new server with v5? I assume there is no upgrade
path ?

**Reply by madeindra** [reply]
There's a lot of upgrades, but I have not yet succeeded in running it

**Comment by afyfe** [comment]
Is video group calling supported in v3 of the signal server components ?

**Reply by madeindra** [reply]
Unfortunately, no

**Comment by afyfe** [comment]
Not succeeded ? Have you tried building a 5.x server? How far have you got
? Which docs are u working off

**Reply by madeindra** [reply]
Still not succeed, I've not got all the dependencies accounts needed for it

**Comment by afyfe** [comment]
Hey, get in touch with me.
